### PR TITLE
Split script into 2 commands: 1 to extract, 1 to filter tools

### PR DESCRIPTION
**Pull request by bebatut**
I am still testing locally the script

### CODE REVIEW

File: bin/extract_galaxy_tools.py [hunk 1]
```diff
@@ -361,34 +342,77 @@ def export_tools(tools: list, output_fp: str) -> None:
     df.to_csv(output_fp, sep="\t", index=False)
 
 
+def filter_tools(tools, ts_cat, excluded_tools, keep_tools):
```
Comment (nsoranzo) [inline bin/extract_galaxy_tools.py:345]:
```suggestion
def filter_tools(tools, ts_cat: List[str], excluded_tools: List[str], keep_tools: List[str]):
```

File: bin/extract_galaxy_tools.py [hunk 2]
```diff
@@ -135,28 +135,23 @@ def check_categories(ts_categories, ts_cat):
     :param ts_cat: list of ToolShed categories to keep in the extraction
     """
     if ts_categories is not None and len(ts_cat) > 0:
```
Comment (nsoranzo) [inline bin/extract_galaxy_tools.py:137]:
```suggestion
    if not ts_cat:
        return True
    if not ts_categories:
        return False
```

File: bin/extract_galaxy_tools.py [hunk 3]
```diff
@@ -135,28 +135,23 @@ def check_categories(ts_categories, ts_cat):
...
+        ts_cats = ts_categories.split(", ")
         to_keep = False
-        for cat in ts_categories:
+        for cat in ts_cats:
             if cat in ts_cat:
                 to_keep = True
         return to_keep
     return True
```
Comment (nsoranzo) [inline bin/extract_galaxy_tools.py:144]:
```suggestion
    return bool(set(ts_cat) & set(ts_cats))
```

File: bin/extract_galaxy_tools.py [hunk 4]
```diff
@@ -135,28 +135,23 @@ def check_categories(ts_categories, ts_cat):
...
 
 
-def get_tool_metadata(tool, repo, ts_cat, excluded_tools, keep_tools):
+def get_tool_metadata(tool: ContentFile, repo: Repository) -> :
```
Comment (nsoranzo) [inline bin/extract_galaxy_tools.py:147]: Sorry, forgot to finish this suggestion:
```suggestion
def get_tool_metadata(tool: ContentFile, repo: Repository) -> Optional[Dict[str, Any]]:
```

and needs also this change at the top:

```diff
diff --git a/bin/extract_galaxy_tools.py b/bin/extract_galaxy_tools.py
index 11cbd2d..9f32804 100644
--- a/bin/extract_galaxy_tools.py
+++ b/bin/extract_galaxy_tools.py
@@ -5,7 +5,12 @@ import base64
 import time
 import xml.etree.ElementTree as et
 from pathlib import Path
-from typing import List
+from typing import (
+    Any,
+    Dict,
+    List,
+    Optional,
+)
 
 import pandas as pd
 import requests
```

File: bin/extract_galaxy_tools.py [hunk 5]
```diff
@@ -134,29 +134,22 @@ def check_categories(ts_categories, ts_cat):
...
-        for cat in ts_categories:
-            if cat in ts_cat:
-                to_keep = True
-        return to_keep
-    return True
+    if not ts_cat:
+        return True
+    if not ts_categories:
+        return False
+        ts_cats = ts_categories.split(", ")
```
Comment (nsoranzo) [inline bin/extract_galaxy_tools.py:141]:
```suggestion
    ts_cats = ts_categories.split(", ")
```

Comment (bebatut) [reply bin/extract_galaxy_tools.py:141]: Stupid me...

File: bin/extract_galaxy_tools.py [hunk 6]
```diff
@@ -336,59 +322,112 @@ def parse_tools(repo: Repository, ts_cat, excluded_tools, keep_tools):
...
+        # parse tools in GitHub repositories to extract metada, filter by TS categories and export to output file
+        tools: List[Dict] = []
+        for r in repo_list:
+            print(r)
+            if "github" not in r:
+                continue
+            repo = get_github_repo(r, g)
+            tools += parse_tools(repo)
+            export_tools(tools, args.all_tools, format_list_col=True)
+            print()
```
Comment (nsoranzo) [inline bin/extract_galaxy_tools.py:424]: I suppose this should go outside of the for loop?

```suggestion
        export_tools(tools, args.all_tools, format_list_col=True)
```

Comment (bebatut) [reply bin/extract_galaxy_tools.py:424]: I put in the loop in case the job stops and we have some data already

Comment (nsoranzo) [reply bin/extract_galaxy_tools.py:424]: Ah, then I'll push a patch to write to a file handle instead.

**Review by nsoranzo — APPROVED** [review]

**Comment by paulzierep** [comment]
Worked for my toy set, also with duplicates, running for all repos just to be sure, then merge

**Pull request merged**